packages/govuk-frontend/src/govuk/components/skip-link/skip-link.test.js

```diff
@@ -81,5 +81,21 @@ describe('Skip Link', () => {
         message: 'GOV.UK Frontend is not supported in this browser'
       })
     })
+
+    it('throws when receiving the wrong type for $module', async () => {
+      await expect(
+        renderAndInitialise(page, 'skip-link', {
+          params: examples.default,
+          beforeInitialisation() {
+            // Remove the root of the components as a way
+            // for the constructor to receive the wrong type for `$module`
+            document.querySelector('[data-module]').remove()
```

Comment:
Should users with `null` selectors see `MissingElementError` instead?

More informative for `TypeError` to throw for the wrong selector element

E.g. Trying to pass a `<button>` into `new CharacterCount()`